File: Algorithms/python/003 - Longest Substring Without Repeating Characters.py
class Solution:
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        n = len(s)
        ans = 0
        index = {}
        i = 0
        
        for j in range(n):
            i = max(index.get(s[j], 0), i)
            ans = max(ans, j - i + 1)
            index[s[j]] = j + 1
    
        return ans 

File: Algorithms/cpp/003 - Longest Substring Without Repeating Characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int index[128] = {0};
        int ans=0, i=0, j=0;
        for(i,j ;j < s.length(); j++){
            i = max(index[s[j]], i);
            ans = max(ans, j - i + 1);
            index[s[j]] = j + 1;
        }
        return ans;
    }
};